src/spaces/homspace.jl

```diff
@@ -349,3 +349,18 @@ function fusionblockstructure(W::HomSpace, ::GlobalLRUCache)
     end
     return structure
 end
+
+# Diagonal ranges
+#----------------
+# TODO: is this something we want to cache?
+function diagonalblockstructure(W::HomSpace)
```

Comment:
If we consider this part of the official interface, we might want to add a test for `W isa HomSpace{S,1,1}` and `W.domain[1] == W.codomain[1]`?